cmd/appgw-ingress/main.go

```diff
@@ -199,18 +215,11 @@ func main() {
 		klog.Fatal(errorLine)
 	}
 
-	// associate route table to application gateway subnet
-	if cpConfig != nil && cpConfig.RouteTableName != "" {
-		subnetID := *(*appGw.GatewayIPConfigurations)[0].Subnet.ID
-		routeTableID := azure.RouteTableID(azure.SubscriptionID(cpConfig.SubscriptionID), azure.ResourceGroup(cpConfig.RouteTableResourceGroup), azure.ResourceName(cpConfig.RouteTableName))
-
-		err = azClient.ApplyRouteTable(subnetID, routeTableID)
-		if err != nil {
-			klog.V(3).Infof("Unable to associate Application Gateway subnet '%s' with route table '%s' due to error (this is relevant for AKS clusters using 'Kubenet' network plugin): [%+v]",
-				subnetID,
-				routeTableID,
-				err)
+	if err := cni.ReconcileCNI(context.Background(), azClient, ctrlClient, env.AGICPodNamespace, cpConfig, appGw); err != nil {
```

Comment:
@nairashu was also suggesting to validate at AGIC level that VNET ID is the same for OverlayExtensionConfig ExtensionIPRange cidr and for aks cluster node ip addresses.

Reply:
Ok. Do you what is the best way to get the Cluster VNET ID on the controller ?

Reply:
Will take this up as a separate pr